buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/measurement/coverage/GenerateMeasurementsTask.groovy

```diff
@@ -68,63 +76,34 @@ public class GenerateMeasurementsTask extends DefaultTask {
         }
     }
 
-    private def trimProjectPathLeadingColon(path) {
-        return path.startsWith(':') ? path.drop(1) : path
+    private def findAllFirebaseProductProjects() {
+        return project.rootProject.allprojects.findAll {
+            it.extensions.findByType(FirebaseLibraryExtension) != null
+        }
+    }
+
+    private def findAllProductCoverageTasks() {
+        return findAllFirebaseProductProjects().collect {
+            it.tasks.withType(JacocoReport)
+        }.flatten()
     }
 
     private def generateJson(pullRequestNumber) {
-        def coverages = [:]
-
-        // TODO(yifany@): Consolidate mappings with apksize and iOS.
-        def sdkMap = [
-                'firebase-common': 0,
-                'firebase-common-ktx': 1,
-                'firebase-database': 2,
-                'firebase-database-collection': 3,
-                'firebase-firestore': 4,
-                'firebase-firestore-ktx': 5,
-                'firebase-functions': 6,
-                'firebase-inappmessaging-display': 7,
-                'firebase-storage': 8,
-                'firebase-datatransport': 9,
-
-                // 'firebase-common-ktx' and 'firebase-firestore-ktx' has been
-                // made a subproject of their corresponding java projects in PR
-                // #409 and #426. Old mappings are kept for backward
-                // compatibility reason.
-                'firebase-common:ktx': 10,
-                'firebase-firestore:ktx': 11
-        ]
-
-        for (Project p: project.rootProject.subprojects) {
-            def name = trimProjectPathLeadingColon(p.path)
-            if (name.startsWith('firebase')) {
-                def percent = getCoveragePercentFromReport(p)
-                if (sdkMap.containsKey(name)) {
-                    coverages[sdkMap[name]] = percent
-                } else {
-                    project.logger.warn("Find SDK with name: $name not defined in the SDK to ID mapping.")
-                }
-            }
-        }
+        def now = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date())
+        def projects = findAllFirebaseProductProjects()
 
-        def replacements = coverages.collect {
-            "[$pullRequestNumber, $it.key, $it.value]"
-        }.join(", ")
+        def replacements = projects.collect {
+            [ it.path, pullRequestNumber, getCoverageForProject(it), now ]
```

Comment:
Should this be the project path or product name?

Reply:
`project.path` gives results like `:firebase-common` or `:firebase-firestore:ktx`.  `project.name` gives `firebase-common` or `ktx` for the same projects respectively. We can assembly the full name based on its name and parent project's name. But I figured using `path` is easier and I'm ok with the formatting.

Reply:
SGTM.